pkg/controller/podgroup_test.go

```diff
@@ -0,0 +1,181 @@
+			podPhase:          v1.PodFailed,
+			previousPhase:     v1alpha1.PodGroupScheduled,
+			desiredGroupPhase: v1alpha1.PodGroupFailed,
+		},
+		{
+			name:              "Group finished",
+			pgName:            "pg3",
+			minMember:         2,
+			podNames:          []string{"pod1", "pod2"},
+			podPhase:          v1.PodSucceeded,
+			previousPhase:     v1alpha1.PodGroupScheduled,
+			desiredGroupPhase: v1alpha1.PodGroupFinished,
+		},
+		{
+			name:              "Group status convert from scheduling to scheduled",
+			pgName:            "pg4",
+			minMember:         2,
+			podNames:          []string{"pod1", "pod2"},
+			podPhase:          v1.PodPending,
+			previousPhase:     v1alpha1.PodGroupScheduling,
+			desiredGroupPhase: v1alpha1.PodGroupScheduled,
+		},
+		{
+			name:              "Group status convert from scheduling to succeed",
+			pgName:            "pg5",
+			minMember:         2,
+			podNames:          []string{"pod1", "pod2"},
+			podPhase:          v1.PodPending,
+			previousPhase:     v1alpha1.PodGroupScheduling,
+			desiredGroupPhase: v1alpha1.PodGroupFinished,
+			podNextPhase:      &podSuccess,
+		},
+		{
+			name:              "Group status convert from scheduling to succeed",
+			pgName:            "pg6",
+			minMember:         2,
+			podNames:          []string{"pod1", "pod2"},
+			podPhase:          v1.PodPending,
+			previousPhase:     v1alpha1.PodGroupScheduling,
+			desiredGroupPhase: v1alpha1.PodGroupFailed,
+			podNextPhase:      &podFailed,
+		},
+		{
+			name:               "Group group should not enqueue, created too long",
+			pgName:             "pg7",
+			minMember:          2,
+			podNames:           []string{"pod1", "pod2"},
+			podPhase:           v1.PodRunning,
+			previousPhase:      v1alpha1.PodGroupPending,
+			desiredGroupPhase:  v1alpha1.PodGroupPending,
+			podGroupCreateTime: &createTime,
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			ps := pods(c.podNames, c.pgName, c.podPhase)
+			kubeClient := fake.NewSimpleClientset(ps[0], ps[1])
+			pg := pg(c.pgName, 2, c.previousPhase, c.podGroupCreateTime)
+			pgClient := pgfake.NewSimpleClientset(pg)
+
+			informerFactory := informers.NewSharedInformerFactory(kubeClient, controller.NoResyncPeriodFunc())
+			schedulingInformerFactory := pgformers.NewSharedInformerFactory(pgClient, controller.NoResyncPeriodFunc())
+			podInformer := informerFactory.Core().V1().Pods()
+			pgInformer := schedulingInformerFactory.Scheduling().V1alpha1().PodGroups()
+			ctrl := NewPodGroupController(kubeClient, pgInformer, podInformer, pgClient)
+
+			informerFactory.Start(ctx.Done())
+			schedulingInformerFactory.Start(ctx.Done())
+			if c.podNextPhase != nil {
+				ps := pods(c.podNames, c.pgName, *c.podNextPhase)
+				for _, p := range ps {
+					kubeClient.CoreV1().Pods(p.Namespace).UpdateStatus(ctx, p, metav1.UpdateOptions{})
+				}
+			}
+			go ctrl.Run(1, ctx.Done())
+			time.Sleep(200 * time.Millisecond)
```

Comment:
Sleep() is a bad smell. If you want to wait for the status to get updated, use `wait.poll()` (or similar logic) to ensure that have happened.

Reply:
Thanks